common-app-lib/src/cmr/common_app/services/kms_fetcher.clj

```diff
@@ -51,66 +52,68 @@
   "The key used to store the KMS cache in the system cache map."
   :kms)
 
+(def kms-cache-ttl
+  "The Time-To-Live value, use nil to never expire."
+  nil)
+
+;; called by system.clj across several applications
 (defn create-kms-cache
   "Used to create the cache that will be used for caching KMS keywords. All applications caching
   KMS keywords should use the same fallback cache to ensure functionality even if GCMD KMS becomes
   unavailable."
   []
   (redis-cache/create-redis-cache {:keys-to-track [kms-cache-key]
                                    :read-connection (redis-config/redis-read-conn-opts)
-                                   :primary-connection (redis-config/redis-conn-opts)}))
+                                   :primary-connection (redis-config/redis-conn-opts)
+                                   :ttl kms-cache-ttl}))
 
+;; Called indirectly by bootstrap. This may call KMS service directly through transmit.
 (defn- fetch-gcmd-keywords-map
   "Calls GCMD KMS endpoints to retrieve the keywords. Response is a map structured in the same way
   as used in the KMS cache."
   [context]
   (try
-    (let [kms-cache (cache/context->cache context kms-cache-key)
-          _ (rl-util/log-redis-reading-start "kms-fetcher.fetch-gcmd-keywords-map" kms-cache-key)
-          [t1 kms-cache-value] (util/time-execution (cache/get-value kms-cache kms-cache-key))]
-      (rl-util/log-redis-read-complete "fetch-gcmd-keywords-map" kms-cache-key t1)
-      (kms-lookup/create-kms-index
-        context
-        (into {}
-              (for [keyword-scheme (keys kms/keyword-scheme->field-names)]
-                ;; if the keyword-scheme-value is nil that means we could not get the KMS keywords
-                ;; in this case use the cached value instead so that we don't wipe out the cache.
-                (if-let [keyword-scheme-value (kms/get-keywords-for-keyword-scheme context keyword-scheme)]
-                  [keyword-scheme keyword-scheme-value]
-                  [keyword-scheme (get kms-cache-value keyword-scheme)])))))
+    (kms-lookup/create-kms-index!
```

Comment:
it is my understanding that ! is used when a function makes a change in place. The more I see this function, the more I feel like ! is justified as it changes Redis.

Reply:
https://github.com/bbatsov/clojure-style-guide?tab=readme-ov-file#unsafe-functions